package org.koreait.tests;

import org.junit.jupiter.api.Test;
import org.koreait.testutils.MockMember;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;

@SpringBootTest
@AutoConfigureMockMvc
public class Ex01 {
    @Autowired
    private MockMvc mockMvc;

    @Test
    //@WithMockUser(authorities = "ADMIN")
    //@WithUserDetails(value="user01@test.org")
    @MockMember
    void test1() throws Exception {
        mockMvc.perform(get("/mypage"))
                .andDo(print());
    }
}
